resolver/resolver.go

```diff
@@ -135,54 +129,15 @@ func (res *Resolver) Serve(proto string) (<-chan struct{}, <-chan error) {
 	return ch, errCh
 }
 
-// LaunchZK launches a Zookeeper detector, returns immediately two chans: the first fires an empty
-// struct whenever there's a new (non-nil) mesos leader, the second if there's an unrecoverable
-// error in the master detector.
-func (res *Resolver) LaunchZK(initialDetectionTimeout time.Duration) (<-chan struct{}, <-chan error) {
-	var startedOnce sync.Once
-	startedCh := make(chan struct{})
-	errCh := make(chan error, 1)
-	leaderCh := make(chan struct{}, 1) // the first write never blocks
-
-	listenerFunc := func(newLeader string) {
-		defer func() {
-			if newLeader != "" {
-				leaderCh <- struct{}{}
-				startedOnce.Do(func() { close(startedCh) })
-			}
-		}()
-		res.leaderLock.Lock()
-		defer res.leaderLock.Unlock()
-		res.leader = newLeader
-	}
-	go func() {
-		defer util.HandleCrash()
-
-		err := res.startZKdetection(res.config.Zk, listenerFunc)
-		if err != nil {
-			errCh <- err
-			return
-		}
-
-		logging.VeryVerbose.Println("Warning: waiting for initial information from Zookeper.")
-		select {
-		case <-startedCh:
-			logging.VeryVerbose.Println("Info: got initial information from Zookeper.")
-		case <-time.After(initialDetectionTimeout):
-			errCh <- fmt.Errorf("timed out waiting for initial ZK detection, exiting")
-		}
-	}()
-	return leaderCh, errCh
+// SetMasters sets the given masters
```

Comment:
would be nice to have a comment about how SetMasters and Reload are not expected to be invoked concurrently